pkg/daemonsetlock/daemonsetlock.go

```diff
@@ -6,11 +6,17 @@ import (
 	"fmt"
 	"time"
 
+	v1 "k8s.io/api/apps/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
+const (
+	k8sAPICallRetrySleep   = 5 * time.Second // How much time to wait in between retrying a k8s API call
```

Comment:
I put 5 second sleeps between retries, and a 5 minute timeout here. Open to changing that if that doesn't seem quite right as a default.

Maybe these could be user-configurable options? But that seems overkill to start.

Reply:
> Maybe these could be user-configurable options? But that seems overkill to start.

I agree; retrying at all is the important bit, reasonable defaults are fine for now.